pip/_vendor/distlib/database.py

```diff
@@ -27,7 +27,7 @@
 
 __all__ = ['Distribution', 'BaseInstalledDistribution',
            'InstalledDistribution', 'EggInfoDistribution',
-           'DistributionPath']
+           'DistributionPath', 'get_required_dists']
```

Comment:
You cannot modify any files in `pip/_vendor`. To do so you'd need to make a PR on the corresponding project.

But I don't think this is needed, you basically need to iterate on all packages from `pkg_resources.working_set` and check if `dist` is in their `requires()`.

Ideally you could also detect if is is required by some extras.